src/lib.rs

```diff
@@ -278,13 +278,15 @@ impl<T> Ioctl<Read, &T> {
     /// internal `ioctl()` call. Upon success, returns the raw (positive)
     /// return value and the instance of `T`.
     pub fn ioctl(self, fd: &impl AsRawFd) -> Result<(c_uint, T)> {
-        let mut out: T = unsafe { core::mem::MaybeUninit::zeroed().assume_init() };
+        use core::mem::MaybeUninit;
 
-        let r = unsafe { ioctl(fd.as_raw_fd(), self.0, &mut out as *mut _, null::<c_void>()) };
+        let mut out: MaybeUninit<T> = MaybeUninit::zeroed();
```

Comment:
I know this is my fault, but do we want zeroed or uninit here?

Reply:
Hmm.. uninit() might be fine, because the kernel should not expect anything in there yet.

Reply:
I think zeroed is only legal if everything inside of `T` can accept zero as a value